source/Platforms.hx

```diff
@@ -0,0 +1,178 @@
+        minY = U;
+        maxY = D;
+        platformwidth = blockSize * W;
+
+		// Initializes a basic graphic for the player
+		makeGraphic(platformwidth, blockSize, FlxColor.YELLOW);
+
+	}
+
+    //Movement as requested by user parameters
+    public function movement():Void
+    {
+		if (DOWN) {
+            if (y >= (startY + maxY)) {
+                UP = true;
+                DOWN = false;
+            }
+            else {
+                velocity.y = platformVelocity;
+            }
+        }
+        if (UP) {
+            if (y <= (startY - minY)) {
+                UP = false;
+                DOWN = true;
+            }
+            else {
+                velocity.y = -platformVelocity;
+            }
+        }
+        if (LEFT) {
+            if (x <= (startX - minX)) {
+                RIGHT = true;
+                LEFT = false;
+            }
+            else {
+                velocity.x = -platformVelocity;
+            }
+        }
+        if (RIGHT) {
+            if (x >= startX + maxX) {
+               RIGHT = false;
+               LEFT = true;
+            }
+            else {
+                velocity.x = platformVelocity;
+            }
+        }
+    }
+
+    // Add or update sprites colliding with platform
+    public function platformObjects(elapsed:Float, player:FlxObject, platform:Platforms):Void
+    {
+        // If there is nothing in the touching platform group, add this object
+        if (countTouching == 0)
+        {  
+            var obj = new PlatformTracker(player, (player.x - platform.x));
+            touchingSprites.add(obj);
+            countTouching++;
+        }
+        else
+        {
+            inContact = false;
+            // Iterate through the members of the group
+            for (i in 0...countTouching)
+            {
+                // If this sprite touching the platform has already been touching the platform
+                if (touchingSprites.members[i].returnBase() == player)
+                {
+                    // Mark that this sprite has been found and was already in contact
+                    inContact = true;
+
+                    // Multiply velocity by elapsed to get the player's movement each frame.
+                    touchingSprites.members[i].trackedOffsetX += player.velocity.x * elapsed;
+
+                    // Update player x position
+                    player.x = platform.x + touchingSprites.members[i].returnOffset();
+                }
+            }
+            // If the end of the group is reached and this object isn't already in contact, add it
+            if (!inContact)
+            {
+                var obj = new PlatformTracker(player, (player.x - platform.x));
+                touchingSprites.add(obj);
+                countTouching++;
+            }
+        }
+    }
+
+    // Remove sprites NOT colliding with platform
+    private function notPlatformObjects(elapsed:Float, sprite:FlxObject, platform:Platforms):Void
```

Comment:
/**
     *  Remove sprites NOT colliding with the platform. If objects are touching the platform, check that 
     *  this sprite is not one of them. If the object is found in the list of sprites touching, 
     *  remove it from the group and set its y acceleration back to Gravity as declared in PlayState.hx
     *  
     *  @param   elapsed is the time passed since last update
     *  @param   sprite is the sprite object not touching the platform
     *  @param   platform is the platform not being touched by the sprite
     *  @param   originalCount is the number of sprites in the group before any changes are made; used to 
     *  prevent changes to loop duration
     */